src/main/scala/mesosphere/marathon/core/health/HealthCheck.scala

```diff
@@ -282,6 +290,19 @@ case class MesosHttpHealthCheck(
 object MesosHttpHealthCheck {
   val DefaultPath = None
   val DefaultProtocol = Protocol.MESOS_HTTP
+  val DefaultIpProtocol = IPv4
+
+  def ipProtocolConversion(ipProtocol: IpProtocol): Protos.HealthCheckDefinition.IpProtocol = ipProtocol match {
+    case IPv4 => Protos.HealthCheckDefinition.IpProtocol.IPv4
+    case IPv6 => Protos.HealthCheckDefinition.IpProtocol.IPv6
+    case _ => throw new IllegalStateException(s"Unsupported protocol in health check: $ipProtocol")
```

Comment:
well the only reason I did it like this is because compiler will give us just warning (by the way I tend to think that it is useful to have match errors escalated as build errors but that is a different story...)

So if someone ignores the warning (and I think it is possible it happens) in runtime it will give much more cryptic exception "IPv6 (of class mesosphere.marathon.core.health.IPv6$)
scala.MatchError: IPv6 (of class mesosphere.marathon.core.health.IPv6$)" than the one I put there...

But I agree that with the solution as it is now, we don't even have a chance to get that warning, so I'll remove it. Thanks...